### PR TITLE
feat: add `findBy*` queries

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Added findBy queries for original queries (ByText, ByPlaceholder, ByDisplayValue, ByTestId) as well as for a11y queries. 

*Note:* in case of `findByTestId` I used custom finding algorithm because current implementation of `getByTestId` is [broken](https://github.com/callstack/react-native-testing-library/issues/270).

### Test plan
Added units tests for all of the new queries.

### CODE REVIEW

File: docs/Queries.md [hunk 1]
```diff
@@ -24,6 +24,17 @@ title: Queries
```
Comment (thymikee) [inline docs/Queries.md:24]: can you rebase to master? `docs` were moved to `website/docs`

File: src/helpers/getByAPI.js [hunk 2]
```diff
@@ -157,6 +157,23 @@ export const getByTestId = (instance: ReactTestInstance) =>
     }
   };
 
+export const fixedGetByTestId = (instance: ReactTestInstance) =>
+  function getByTestIdFn(testID: string) {
+    try {
+      const results = getAllByTestId(instance)(testID);
+      if (results.length === 1) return results[0];
+      else
```
Comment (thymikee) [inline src/helpers/getByAPI.js:165]: can we use some old fashioned curly braces with this if statement? :)

File: docs/Queries.md (hunk 1 above)
Comment (mdjastrzebski) [reply docs/Queries.md]: Done :-)

**Comment by mdjastrzebski** [comment]
> Missing updates to `typings/index.d.ts` for TS users, but from a quick look it looks good!

I've added typescipt type definitions & tests for that.

File: typings/index.d.ts [hunk 3]
```diff
@@ -57,61 +71,147 @@ export interface QueryByAPI {
...
-type GetReturn = ReactTestInstance;
-type GetAllReturn = Array<ReactTestInstance>;
-type QueryReturn = ReactTestInstance | null;
-type QueryAllReturn = Array<ReactTestInstance> | [];
+export interface WaitForOptions {
+  timeout: number;
+  interval: number;
+}
+
+export interface FindByAPI {
```
Comment (thymikee) [inline typings/index.d.ts:91]: I don't think we need to export this interface, what for?

Comment (mdjastrzebski) [reply typings/index.d.ts:91]: Same happens for GetByAPI & QueryByAPI interfaces. Do you things it's good idea to remove all of these exports?

Comment (thymikee) [reply typings/index.d.ts:91]: If we don't use them, then let's remove. If they're useful for somebody, they'll send a PR to export it :)

Comment (mdjastrzebski) [reply typings/index.d.ts]: I've removed exports for all three of them

**Comment by mdjastrzebski** [comment]
Reversed prettier run on typings files in order to remove noise from the PR :-)

**Comment by mdjastrzebski** [comment]
I've added overlapping parts with `within` operator which has been merged to master.

File: src/__tests__/a11yAPI.test.js [hunk 4]
```diff
@@ -62,74 +64,127 @@ function Section() {
...
   );
   const button = queryByA11yLabel(/button/g);
   expect(button && button.props.accessibilityLabel).toEqual(BUTTON_LABEL);
+
+  const asyncButton = await findByA11yLabel(BUTTON_LABEL);
+  expect(asyncButton.props.accessibilityLabel).toEqual(BUTTON_LABEL);
+
   expect(() => getByA11yLabel(NO_MATCHES_TEXT)).toThrow(NO_INSTANCES_FOUND);
   expect(queryByA11yLabel(NO_MATCHES_TEXT)).toBeNull();
+  await expect(
```
Comment (thymikee) [inline src/__tests__/a11yAPI.test.js:83]: Can we move all async queries to the bottom of these tests? There's a possibility that a sync query would regress and doing async stuff in-between could hide that.

File: website/docs/Queries.md [hunk 5]
```diff
@@ -24,6 +24,17 @@ title: Queries
...
+### findBy
+
+`findBy` queries return a promise which resolves when a matching element is found. The promise is rejected if no elements match or if more than one match is found after a default timeout of 4500ms. If you need to find more than one element, then use `findAllBy`.
+
+### findAllBy
+
+`findAllBy` queries return a promise which resolves to an array when any matching elements are found. The promise is rejected if no elements match after a default timeout of 4500ms.
+
+
+*Note:* `findBy` and `findAllBy` queries accept optional `waitForOptions` object argument which can contain `timeout` and `interval` properies which have the same meaning as respective arguments to [`waitForElement`](https://callstack.github.io/react-native-testing-library/docs/api#waitforelement) function.
```
Comment (thymikee) [inline website/docs/Queries.md:36]:
```suggestion
:::info
`findBy` and `findAllBy` queries accept optional `waitForOptions` object argument which can contain `timeout` and `interval` properies which have the same meaning as respective arguments to [`waitForElement`](https://callstack.github.io/react-native-testing-library/docs/api#waitforelement) function. 
:::
```

File: src/__tests__/a11yAPI.test.js (hunk 4 above)
Comment (mdjastrzebski) [reply src/__tests__/a11yAPI.test.js:86]: Done

**Pull request merged**